### PR TITLE
[synthetics-private-location] Fix commonLabels duplicated in Deployment

### LINKED ISSUE #840: [synthetics-private-location] `commonLabels` duplicated in Deployment
**Describe what happened:**
When setting some `commonLabels` in Values, the deployment of the chart fails.

Example in the _values.yaml_:
```yaml
commonLabels:
  team_name: dev
```

That produce: 
```yaml
# Source: synthetics-private-location/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: synthetics-private-location
  labels:
    helm.sh/chart: synthetics-private-location-0.15.5
    app.kubernetes.io/name: synthetics-private-location
    app.kubernetes.io/instance: synthetics-private-location
    **team_name: dev**
    app.kubernetes.io/version: "1.29.0"
    app.kubernetes.io/managed-by: Helm
    **team_name: dev**
```

**Describe what you expected:** 

I expected that `commonLabels` were not duplicated because it makes my Flux HelmRelease fail.

**Steps to reproduce the issue:**

Create a `./values_test.yaml` file and copy the example provided in `Describe what happened` section . Then, run `helm template synthetics-private-location ./synthetics-private-location -f ./values_test.yaml`

### PR DESCRIPTION
**Pull request by martinpiegay**
#### What this PR does / why we need it: 
Fix commonLabels duplicated in Deployment

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #840 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

**Review by loutPhilipps — APPROVED** [review]

**Comment by loutPhilipps** [comment]
Thank you for the fix!

**Comment by martinpiegay** [comment]
Thanks for the review @loutPhilipps.
I updated the Chart.yaml and CHANGELOG.md.

File: charts/synthetics-private-location/templates/_helpers.tpl [hunk 1]
```diff
@@ -40,9 +40,6 @@ helm.sh/chart: {{ include "synthetics-private-location.chart" . }}
 app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
 {{- end }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
-{{- if .Values.commonLabels}}
-{{ toYaml .Values.commonLabels }}
-{{- end }}
```
Comment (mzneos) [inline charts/synthetics-private-location/templates/_helpers.tpl:45]: Shouldn't this be removed below, in the `Selector labels` part? 
https://github.com/DataDog/helm-charts/blob/07f38ede48cd01b7ba18000c622eccaeb143d035/charts/synthetics-private-location/templates/_helpers.tpl#L54-L56

Comment (loutPhilipps) [reply charts/synthetics-private-location/templates/_helpers.tpl:45]: If we remove commonLabels from selector labels, they won't be used for [selectors](https://github.com/DataDog/helm-charts/blob/611690287a32951452dc0d9bc014d9ffb3265d58/charts/synthetics-private-location/templates/deployment.yaml#L11) anymore, so I think this PR is removing the right template block

File: charts/synthetics-private-location/CHANGELOG.md [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
 
-## 0.15.7
+## 0.15.8
+
+* Fix commonLabels duplicated in Deployment.
 
+## 0.15.7
 * Update private location image version to `1.31.1`.
 
+
```
Comment (loutPhilipps) [inline charts/synthetics-private-location/CHANGELOG.md:10]:
```suggestion
```

**Review by loutPhilipps — COMMENTED** [review]
Small nitpick

**Comment by martinpiegay** [comment]
@loutPhilipps can you merge this PR now ?
Thanks

**Comment by loutPhilipps** [comment]
> @loutPhilipps can you merge this PR now ? Thanks

Looks like some of your earlier commits are not signed. I cannot merge this PR otherwise. You can rebase this PR to rewrite the history

File: charts/synthetics-private-location/CHANGELOG.md [hunk 3]
```diff
@@ -1,7 +1,10 @@
 # Datadog changelog
 
-## 0.15.7
+## 0.15.8
+
+* Fix commonLabels duplicated in Deployment.
 
+## 0.15.7
 * Update private location image version to `1.31.1`.
```
Comment (loutPhilipps) [inline charts/synthetics-private-location/CHANGELOG.md:8]:
```suggestion

* Update private location image version to `1.31.1`.
```
Missing a newline here

**Comment by martinpiegay** [comment]
@loutPhilipps done

File: charts/synthetics-private-location/CHANGELOG.md [hunk 4]
```diff
@@ -1,12 +1,16 @@
...
+* Fix commonLabels duplicated in Deployment.
+
 ## 0.15.7
 
 * Update private location image version to `1.31.1`.
 
 ## 0.15.6
 
-* Update private location image version to `1.31.0`.
+* Fix commonLabels duplicated in Deployment.
```
Comment (loutPhilipps) [inline charts/synthetics-private-location/CHANGELOG.md:13]: You've overwritten a previous version Changelog
```suggestion
* Update private location image version to `1.31.0`.
```

Comment (martinpiegay) [reply charts/synthetics-private-location/CHANGELOG.md]: indeed, sorry.

**Review by loutPhilipps — APPROVED** [review]

**Review by loutPhilipps — COMMENTED** [review]
Last thing - you need to update the `README.md` to change the chart version at the top:
```diff
# Datadog Synthetics Private Location

- ![Version: 0.15.7](https://img.shields.io/badge/Version-0.15.7-informational?style=flat-square) ![AppVersion: 1.31.1](https://img.shields.io/badge/AppVersion-1.31.1-informational?style=flat-square)
+ ![Version: 0.15.8](https://img.shields.io/badge/Version-0.15.8-informational?style=flat-square) ![AppVersion: 1.31.1](https://img.shields.io/badge/AppVersion-1.31.1-informational?style=flat-square)
```

**Comment by martinpiegay** [comment]
@loutPhilipps done

**Pull request merged**